lib/client.ml

```diff
@@ -1,40 +1,49 @@
 open Lwt_unix
-open Util
 
-type 'a t =
-  { address: string
-  ; port: int
-  ; socket: file_descr
-  ; state: 'a ref
-  ; recv_mutex: Lwt_mutex.t
-  ; state_mutex: Lwt_mutex.t
-  }
+type 'a t = {
+  address : string;
+  port : int;
+  socket : file_descr;
+  state : 'a ref;
+  recv_mutex : Lwt_mutex.t;
+  state_mutex : Lwt_mutex.t;
+}
 
-let peer_from (client: 'a t) =
-  Peer.{ address = client.address; port = client.port}
+let peer_from (client : 'a t) =
+  Peer.
+    {
+      address = client.address;
+      port = client.port;
+      known_peers = [];
+      state = Alive;
+    }
 
 let send_to client payload peer =
   let len = Bytes.length payload in
   let addr = Peer.to_sockaddr peer in
   let%lwt _ = sendto !client.socket payload 0 len [] addr in
   Lwt.return ()
 
+let naive_broadcast client payload peers =
+  let _ = List.map (send_to client payload) peers in
+  Lwt.return ()
+
 let recv_next client =
   let open Lwt_unix in
-  (* Peek at the first 8 bytes of the incoming datagram
-  to read the Bin_prot size header. *)
-  let size_buffer = Bytes.create 8 in
+  let open Util.Encoding in
```

Comment:
In my opinion (which is not necessarily the rule), it's nice to open up till the last module (e.g. when using functions from Util.Encoding, only do `let open Util in ...`) so that it's obvious where we call functions that weren't defined in this file. In other words, any function from `Util.Encoding` have `Encoding.` before them, so it's obvious they're not from this module. Otherwise, you have to check via the editor features which are not present in GitHub for example. What do you think?

Reply:
Agreed. I feel like it is more explicit to do this way. Thanks!